### DISCUSSION TITLE
define extras_require in setup.py

### DISCUSSION THREAD

**Discussion by gnikk** [pypa/setuptools#2706] [Q&A] [answered]
First I defined this dictionary and its keys, then If I have a spell problem in writing keys, I won't get an error and everything seem successfully.
Is there any solution for warning this error?
and for the default option, I want to install my package without any parameter. Is it possible?

**Comment by webknjaz** [comment, marked as answer]
Could you please add more details and examples? I'm not sure what you're asking. FWIW Pip warns about non-existent extras. Also, normal `install_requires` are always pulled in by pip when it resolves the deps.
